### PR TITLE
Add API endpoints for jobs and companies

### PR DESCRIPTION
**Pull request by EmilyMB**
Added:
- Jobs Index endpoint
- Company serializer
- Company show and index endpoints
- Tag serializer
- Tag index endpoint

### CODE REVIEW

File: spec/controllers/api/v1/companies_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      3.times { create(:company) }
+    end
+
+    it "#GET api/v1/companies" do
+      get :index
+      result = JSON.parse(response.body)["companies"]
+
+      expect(response.status).to eq(200)
+      expect(result.count).to eq(3)
+      expect(result).to have_content("Twitter")
```
Comment (mikedao) [inline spec/controllers/api/v1/companies_controller_spec.rb:28]: Great checking for response and content.

File: spec/controllers/api/v1/jobs_controller_spec.rb [hunk 2]
```diff
@@ -78,4 +78,20 @@
...
+    it "#GET api/v1/jobs" do
+      get :index
+      results = JSON.parse(response.body)["jobs"]
+
+      expect(response.status).to eq(200)
+      expect(results.count).to eq(5)
+      expect(results).to have_content("Job 24")
+    end
+  end
 end
```
Comment (mikedao) [inline spec/controllers/api/v1/jobs_controller_spec.rb:108]: No sad path tests?

Comment (EmilyMB) [inline spec/controllers/api/v1/jobs_controller_spec.rb]: OK, let me add some.

**Comment by mikedao** [comment]
:sparkling_heart:

**Pull request merged**